realm/realm-annotations-processor/src/main/java/io/realm/processor/Utils.java

```diff
@@ -151,6 +151,13 @@ public static boolean isRealmObject(VariableElement field) {
     }
 
     /**
+     * @return {@code ture} if a given field type is nullable PrimaryKey type, {@code false} otherwise.
+     */
+    public static boolean isNullablePrimaryKeyType(VariableElement field) {
+        return Constants.JAVA_TO_NULLABLE_PRIMARYKEY_TYPES.contains(field.asType().toString());
```

Comment:
See above.  I don't think we need to maintain an array of these. Just check if the field is a Boxed type and a primary key field instead.